homeassistant/components/generic/camera.py

```diff
@@ -85,15 +89,19 @@ def __init__(self, hass, device_info):
         self._supported_features = SUPPORT_STREAM if self._stream_source else 0
         self.content_type = device_info[CONF_CONTENT_TYPE]
         self.verify_ssl = device_info[CONF_VERIFY_SSL]
+        if device_info.get(CONF_RTSP_TRANSPORT):
+            self.stream_options[FFMPEG_OPTION_MAP[CONF_RTSP_TRANSPORT]] = device_info[
```

Comment:
This looks equivalent to `self.stream_options[CONF_RTSP_TRANSPORT] = device_info[CONF_RTSP_TRANSPORT]`.  Is the double look up here serve another purpose?

Reply:
Yeah it is equivalent. It's to decouple what we are using for in HA for `CONF_RTSP_TRANSPORT` from what pyav/ffmpeg uses as an input option. Both happen to be `rtsp_transport`. The code in `onvif` did not decouple these.
If we wanted to be anal, I guess we could decouple the values as well :-P

Reply:
Makes sense.

Sorry also I just saw this was a draft.  I cam across it just looking at the review backlog and didn't notice.